onnxruntime/test/util/compare_ortvalue.cc

```diff
@@ -65,6 +66,53 @@ const char* ElementTypeToString(MLDataType type) {
   return DataTypeImpl::ToString(type);
 }
 
+template <typename T>
+std::pair<COMPARE_RESULT, std::string> CheckCosineSimilarity(const Tensor& outvalue, const Tensor& expected_value) {
+  const size_t tensor_size = static_cast<size_t>(expected_value.Shape().Size());
+  const T* expected_output = expected_value.Data<T>();
+  const T* real_output = outvalue.Data<T>();
+  std::pair<COMPARE_RESULT, std::string> res = std::make_pair(COMPARE_RESULT::SUCCESS, "");
+  const float cosine_similarity_threshold = 0.99;
+
+  T dot = 0.0, denom_a = 0.0, denom_b = 0.0;
```

Comment:
Add 'f' to each of the constants. Like, 0.0f. Without of the suffix the literals would be in double type instead of float. Float values can be implicitly converted to double, but not vice versa.  Or you may add a static_cast here.

Reply:
thanks, @snnn . I have fixed and pushed a new version.